### DISCUSSION TITLE
Announcing: WinUI 3 in Windows App SDK 1.1 🔃✨🔔

### DISCUSSION THREAD

**Discussion by gabbybilka** [microsoft/microsoft-ui-xaml#7394] [Announcements]
### WinUI 3 in Windows App SDK 1.1.0

You are now able to use the stable version of Windows App SDK 1.1 in your applications! This release includes **new features *and* stability & performance improvements** across the component areas for WinUI 3 apps. These include: 

- Apps can create **multiple windows** with WinUI3 content on the same UI thread. 
- **Mica** & **Background Acrylic** are available. 
- **Improved default and custom title bar** with new functionalities like z-order, sizing/resizing, tall title bar, and extended show. 
- Apps using the Windows App SDK can **run with elevated permissions**. 
- Apps that deploy with all their dependencies together are now able to include the Windows App SDK as well.  **(self-contained deployment)** 
- Apps can **push notifications** regardless of being packaged/unpackaged, or whether the app is published in the Store. 
- The **environment variables** PATH, and PATHEXT can be managed and tracked at runtime using new convenient APIs.   
- Apps can **restart** manually or register themselves to be restarted if it was running when a system update occurs. 

**To see everything that's new and changed, see the full [Windows App SDK 1.1 release notes](https://aka.ms/windowsappsdk/1.1/1.1.0/release-notes).**  

### Getting Started 

To get started using Windows App SDK to develop Windows apps, check out the following documentation:    
- [Install developer tools ](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment).     
- [Create a WinUI 3 app](https://docs.microsoft.com/windows/apps/winui/winui3/create-your-first-winui3-app).   
- [Continue your development journey](https://docs.microsoft.com/windows/apps/develop).   

### What's next  

To keep up with progress being made on the Windows App SDK, please see our [feature roadmap](https://aka.ms/winappsdkroadmap).

**Comment by jschwizer99** [comment]
We would like to switch our application from 1.0.3 App SDK to App SDK 1.1.0. But the hard crashes related to drag & drop are for us a hard blocker. See #7002, #7007, #7192. We would have to switch off drag & drop functionality to ship it to customers. I would appreciate it a lot if these instabilities could be addressed with high priority.

**Comment by gabbybilka** [comment]
> We would like to switch our application from 1.0.3 App SDK to App SDK 1.1.0. But the hard crashes related to drag & drop are for us a hard blocker. See #7002, #7007, #7192. We would have to switch off drag & drop functionality to ship it to customers. I would appreciate it a lot if these instabilities could be addressed with high priority.

@codendone, @bpulliam and team have been working on fixes for multiple D&D caused crashes in WinAppSDK and we're planning to include them in servicing releases across the versions (0.8.10, 1.0.4, 1.1.1).

**Comment by jschwizer99** [comment]
@gabbybilka, many thanks for the update. I'm looking forward to getting a working D&D that is again on a level of 1.0.3 or better. If your team needs further help to verify these fixes on the existing applications reach out to the reporters. From the different reports, it looks that D&D suffers from a series of instabilities that may need to be addressed separately.

**Comment by shane-vorwerk** [comment]
@gabbybilka  Do you have a date that the Drag and Drop crash detailed in  #7007 will be fixed? It is I am aware this issue has been present since the SDK was called Project Reunion - see #4574. 

This is becoming urgent from a business perspective as I cannot provide customers with an application that relies on Drag and Drop for one of its core features, that just crashes. (Issues like this have far reaching consequences beyond whether a program is stable or not) If there are any work arounds or suggestions from the team, please could you provide documentation.   

It is good to know that the team is working on this. I hope that you are able to release the fixes very soon.  Also to echo @jschwizer99  if your team needs assistance to actually solve the issues contact me or others who have reported this. It is in all our interests to solve this as quickly and completely as possible.

**Comment by mlynch** [comment]
Would really love to use this but it looks like WebView2 in WinUI3 + Windows App SDK is still broken in unpackaged apps (https://github.com/microsoft/microsoft-ui-xaml/issues/7074)

Hoping that gets resolved soon, had to have customer use WPF instead for their unpackaged use case.

**Comment by wbokkers** [comment]
@shane-vorwerk I do not see many drag/drop crashes in App SDK 1.0.3 with Windows 11. Users using Windows 10 still report drag/drop crashes. In App SDK 1.1 I see only crashes when dragging from outside the app onto the app.

**Comment by shane-vorwerk** [comment]
> @shane-vorwerk I do not see many drag/drop crashes in App SDK 1.0.3 with Windows 11. Users using Windows 10 still report drag/drop crashes. In App SDK 1.1 I see only crashes when dragging from outside the app onto the app.

@wbokkers I can consistently recreate the crash described in #7007 in all versions from 1.0 to 1.1.0 -  the number of times you need to repeat the DnD operation varies between 1 and about 60 repetitions, usually closer to 60. So keep Dragging and Dropping until it crashes, on your system it may take more than 60 repetitions. Use the code referenced in #7007.  I have tested this in both my test program and the  application I am developing. I hope that the Microsoft team can solve all these crashes

**Comment by llothar** [comment]
> Would really love to use this but it looks like WebView2 in WinUI3 + Windows App SDK is still broken in unpackaged apps (#7074) 

I can confirm this. First it only happened when i had 2 WebView2 Views created (what a crazy edge case, isn't it, no engineer could come up with this idea when all they do is unit testing and not app testing). But now even with one of them i run into crashs and just nothing showing up.  And it's not only WinUI3 but also when using it with Win32.

**Comment by lhak** [comment]
Self-contained is a nice feature but in general this is an extremely disappointing release:

- SurfaceImageSource still fills up the gpu memory (used by win2d CanvasControl, https://github.com/microsoft/microsoft-ui-xaml/issues/3665, reported more than 1.5 years ago).
- SwapChains are not usable because they disappear depending on their size (https://github.com/microsoft/microsoft-ui-xaml/issues/6919)
- Flyout shadows still flicker when opened (https://github.com/microsoft/microsoft-ui-xaml/issues/6206)

The advertised mica background functionality also does not work with swapchains (https://github.com/microsoft/microsoft-ui-xaml/issues/7056) and with many controls as it breaks all acrylic brushes (https://github.com/microsoft/microsoft-ui-xaml/issues/7200)

Please note that these all all regressions compared to UWP or using XAML islands.

**Comment by castorix** [comment]
> * SwapChains are not usable because they disappear depending on their size ([Output of swap chain panel is distorted if it gets scaled to a small size #6919](https://github.com/microsoft/microsoft-ui-xaml/issues/6919))

You can use SwapChains without SwapChainPanel
I use them with Direct2D or Media Engine and they work fine (not with the main window handle since this update which adds a black/white background over (apparently as workaround for the flickering problem, but not a very good idea...))

**Comment by shelllet** [comment]
When to fix these bugs?

my complex app doesn't work when upgrading to 1.1 from 1.0.3.


microsoft/microsoft-ui-xaml#6999
microsoft/microsoft-ui-xaml#7191

**Comment by fredemmott** [comment]
> self-contained deployment

Thanks so much for this :) This unblocks me moving from MSIX to classic MSI, removing the need for me to work around the sandbox. I'd be sticking with MSIX if `unvirtualizedResources` and `modifiableApp` capabilities worked with App Installer - i.e. do not require Store or Powershell installation.

**Comment by nikolayvpavlov** [comment]
Window still lacks essential features, that now all require interop to Win32:
1. Icon
2. Query close
3. Sleep / return from sleep
4. Focused / Unfocused.
5. Minimize to tray
6. Minimized / maximized

**Comment by castorix** [comment]
> Window still lacks essential features, that now all require interop to Win32:
> 
> 1. Icon
> 2. Query close
> 3. Sleep / return from sleep
> 4. Focused / Unfocused.
> 5. Minimize to tray
> 6. Minimized / maximized

Icon, intercepting closing, maximizing/minimizing exist (AppWindow, OverlappedPresenter)

**Comment by terrycox** [comment]
jschwizer99's comment regarding drag and drop issues (7002,  7007, and 7192 aren't an exhaustive list of drag and drop issues. This was mentioned in the latest community call. We've added on involving an odd error using a [Treeview with two roots] (https://github.com/microsoft/microsoft-ui-xaml/issues/7266)
